niworkflows/interfaces/masks.py

```diff
@@ -192,6 +194,204 @@ def _post_run_hook(self, runtime):
+
+
+class PrepareRegistrationImagesOutputSpec(TraitedSpec):
+    fixed_file = File(exists=True)
+    moving_file = File(exists=True)
+    fixed_mask = File
+    moving_mask = File
+
+
+class PrepareRegistrationImages(SimpleInterface):
+    input_spec = PrepareRegistrationImagesInputSpec
+    output_spec = PrepareRegistrationImagesOutputSpec
+
+    def _run_interface(self, runtime):
+        fixed_file = self.inputs.fixed_file
+        moving_file = self.inputs.moving_file
+        fixed_mask = self.inputs.fixed_mask
+        moving_mask = self.inputs.moving_mask
+        lesion_mask = self.inputs.lesion_mask
+        explicit_masking = self.inputs.explicit_masking
+
+        # Default behavior: pass-through
+        self._results = {
+            'fixed_file': fixed_file,
+            'moving_file': moving_file,
+            }
+
+        if explicit_masking:
+            if fixed_mask:
+                self._results['fixed_file'] = apply_mask(fixed_file, fixed_mask, cwd=runtime.cwd)
+            if moving_mask:
+                self._results['moving_file'] = apply_mask(moving_file, moving_mask,
+                                                          cwd=runtime.cwd)
+        else:
+            self._results.update({
+                'fixed_mask': fixed_mask,
+                'moving_mask': moving_mask,
+                })
+
+        if lesion_mask:
+            self._results['fixed_mask'] = create_cfm(
+                fixed_mask or fixed_file,
+                lesion_mask=None,
+                global_mask=True,
+                cwd=runtime.cwd)
+            self._results['moving_mask'] = create_cfm(
+                moving_mask or moving_file,
+                lesion_mask=lesion_mask,
+                global_mask=explicit_masking or not moving_mask,
+                cwd=runtime.cwd)
+
+        return runtime
+
+
+def apply_mask(in_file, mask_file, out_filename=None, cwd=None):
+    """
+    Apply a binary mask to an image.
+
+    Parameters
+    ----------
+    in_file : str
+        Path to an image to mask
+    mask_file : str
+        Path to a binary mask
+    out_filename : str (optional)
+        Base filename for output file - if ``None``, ``in_file`` is used
+    cwd : str (optional)
+        Working directory to create CFM in - if ``None``, ``os.getcwd()`` is used
+
+    Returns
+    -------
+    str
+        Absolute path of the masked output image.
+
+    Notes
+    -----
+    in_file and mask_file must be in the same
+    image space and have the same dimensions.
+    """
+    if cwd is None:
+        cwd = os.getcwd()
+
+    out_filename = fname_presuffix(out_filename or in_file, suffix='_masked', newpath=cwd)
+
+    img = nb.load(in_file)
+    mask = nb.load(mask_file)
+    try:
+        _check_space(img, mask)
+    except ValueError as err:
+        msg = err.args[0]
+        raise ValueError(msg + 'Files: {}, {}'.format(in_file, mask_file))
+    data = img.get_data() * (mask.get_data() > 0)
```

Comment:
I feel it's safer to keep the original array and just set zeros outside the mask:
```suggestion
data = img.get_data()
data[mask.get_data() <= 0] = 0
```

Reply:
I don't understand why, but I can do this, if it matters.